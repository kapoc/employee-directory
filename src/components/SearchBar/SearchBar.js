import React from 'react';
import store from '../../store/store';
import { filterEmployees } from '../../actions/actions';

function SearchBar() {
  function onChange(event) {
    store.dispatch(filterEmployees(event.target.value));
  }

  return (
    <div className="form-group" id='searchbar-wrapper'>
      <input type="text" className="form-control" placeholder="Search employees" onChange={onChange}/>
    </div>
  );
}

export default SearchBar;
