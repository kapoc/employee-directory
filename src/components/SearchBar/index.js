import React from 'react';
import store from '../../store/store';
import { setSearchQuery } from '../../actions/actions';

class SearchBar extends React.Component {
  constructor(props){
    super(props);
  }

  onChange = (event) => {
    store.dispatch(setSearchQuery(event.target.value));
  }

  render(){
    return (
      <div className="form-group">
        <input type="text" className="form-control" placeholder="Search employees" onChange={this.onChange}/>
      </div>
    );
  }
}

export default SearchBar;
