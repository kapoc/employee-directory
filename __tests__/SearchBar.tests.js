import React from 'react';
import SearchBar from '../src/components/SearchBar/SearchBar';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/react'


describe('SearchBar tests', () => {
  it('search bar should be rendered with no value', () => {
    const component = renderer.create(
      <SearchBar />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let testInstance = component.root;

    const mockChange = jest.fn()
    const input = testInstance.find(node => node.type === 'input');

    expect(input).not.toBe(null);
    expect(input.value).toEqual(undefined);
  });
});
