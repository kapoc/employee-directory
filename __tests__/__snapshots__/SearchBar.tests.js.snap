// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests search bar should be rendered with no value 1`] = `
<div
  className="form-group"
  id="searchbar-wrapper"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search employees"
    type="text"
  />
</div>
`;
